### PR TITLE
[IE TEST] Continue run after crash

### PR DESCRIPTION
**Pull request by sbalandi**
### Details:
*Now, if crash happens in run stage, it is handled, IE_THROW is thrown and tests continue to run*

### Tickets:
 - *CVS-72076*

### CODE REVIEW

**Comment by sbalandi** [comment]
Conformance run results: https://openvino-ci.intel.com/job/private-ci/job/developer-services/job/e2e/2196/ , https://openvino-ci.intel.com/job/private-ci/job/developer-services/job/e2e/2195/

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/read_ir_test/read_ir.cpp [hunk 1]
```diff
@@ -47,111 +49,124 @@ std::string ReadIRTest::getTestCaseName(const testing::TestParamInfo<ReadIRParam
...
-                    } else {
-                        info.min = -10;
-                    }
-                    if (std::strcmp(p.attribute("max").as_string(), "undefined") != 0) {
-                        info.max = p.attribute("max").as_double();
-                    } else {
-                        info.max = 10;
+    auto crashHandler = std::unique_ptr<CommonTestUtils::CrashHandler>(new CommonTestUtils::CrashHandler());
+
+#ifdef _WIN32
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/read_ir_test/read_ir.cpp:87]: Please, add comments with description

File: src/tests/functional/shared_test_classes/include/shared_test_classes/base/ov_subgraph.hpp [hunk 2]
```diff
@@ -27,9 +27,7 @@ class SubgraphBaseTest : public CommonTestUtils::TestsCommon {
     virtual void query_model();
 
     void TearDown() override {
-        if (!configuration.empty()) {
-            ov::test::utils::PluginCache::get().core().reset();
-        }
+        ov::test::utils::PluginCache::get().core().reset();
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/base/ov_subgraph.hpp:30]: Please try to remove the condition

File: src/tests/ie_test_utils/common_test_utils/test_common.hpp [hunk 3]
```diff
@@ -20,4 +26,10 @@ class TestsCommon : virtual public ::testing::Test {
     std::string GetTestName() const;
 };
 
+class CrashHandler {
```
Comment (iefode) [inline src/tests/ie_test_utils/common_test_utils/test_common.hpp:29]: Please move to separated file

Comment (sbalandi) [reply src/tests/ie_test_utils/common_test_utils/test_common.hpp]: moved

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/read_ir_test/read_ir.cpp (hunk 1 above)
Comment (sbalandi) [reply src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/read_ir_test/read_ir.cpp:93]: done

**Review by iefode — APPROVED** [review]

**Comment by sbalandi** [comment]
last force-pushed was for rebase

**Pull request merged**